### ISSUE TITLE
ERROR: ReferenceError: navigator is not defined

### ISSUE THREAD

**Issue by mobeigi** [cedx/akismet.js#20]
Hey mate,

Great package, thanks for maintaining it!

When using the package from a server side context with a basic example I get:
```
[14:30:56] ERROR: ReferenceError: navigator is not defined
    at new Client (webpack-internal:///(rsc)/./node_modules/@cedx/akismet/lib/client.js:52:50)
    at validationHook (webpack-internal:///(rsc)/./src/payload/collections/Comments/hooks/validationHook.ts:156:35)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async eval (webpack-internal:///(rsc)/./node_modules/payload/dist/collections/operations/create.js:117:20)
    at async createOperation (webpack-internal:///(rsc)/./node_modules/payload/dist/collections/operations/create.js:115:9)
    at async Object.create (webpack-internal:///(rsc)/./node_modules/@payloadcms/next/dist/routes/rest/collections/create.js:25:15)
    at async eval (webpack-internal:///(rsc)/./node_modules/@payloadcms/next/dist/routes/rest/index.js:491:19)
    at async C:\Git\mobeigi.com\app\node_modules\next\dist\compiled\next-server\app-route.runtime.dev.js:6:126975
    at async AppRouteRouteModule.execute (C:\Git\mobeigi.com\app\node_modules\next\dist\compiled\next-server\app-route.runtime.dev.js:6:98611)
    at async AppRouteRouteModule.handle (C:\Git\mobeigi.com\app\node_modules\next\dist\compiled\next-server\app-route.runtime.dev.js:6:128587)
    at async doRender (C:\Git\mobeigi.com\app\node_modules\next\dist\server\base-server.js:1464:42)
    at async responseGenerator (C:\Git\mobeigi.com\app\node_modules\next\dist\server\base-server.js:1798:28)
    at async DevServer.renderToResponseWithComponentsImpl (C:\Git\mobeigi.com\app\node_modules\next\dist\server\base-server.js:1842:28)
    at async DevServer.renderPageComponent (C:\Git\mobeigi.com\app\node_modules\next\dist\server\base-server.js:2215:24)
    at async DevServer.renderToResponseImpl (C:\Git\mobeigi.com\app\node_modules\next\dist\server\base-server.js:2253:32)
    at async DevServer.pipeImpl (C:\Git\mobeigi.com\app\node_modules\next\dist\server\base-server.js:967:25)
    at async NextNodeServer.handleCatchallRenderRequest (C:\Git\mobeigi.com\app\node_modules\next\dist\server\next-server.js:281:17)
    at async DevServer.handleRequestImpl (C:\Git\mobeigi.com\app\node_modules\next\dist\server\base-server.js:860:17)
    at async C:\Git\mobeigi.com\app\node_modules\next\dist\server\dev\next-dev-server.js:377:20
    at async Span.traceAsyncFn (C:\Git\mobeigi.com\app\node_modules\next\dist\trace\trace.js:157:20)
    at async DevServer.handleRequest (C:\Git\mobeigi.com\app\node_modules\next\dist\server\dev\next-dev-server.js:374:24)
    at async invokeRender (C:\Git\mobeigi.com\app\node_modules\next\dist\server\lib\router-server.js:183:21)
    at async handleRequest (C:\Git\mobeigi.com\app\node_modules\next\dist\server\lib\router-server.js:360:24)
    at async requestHandlerImpl (C:\Git\mobeigi.com\app\node_modules\next\dist\server\lib\router-server.js:384:13)
    at async Server.requestListener (C:\Git\mobeigi.com\app\node_modules\next\dist\server\lib\start-server.js:142:13)
```
    
 This is because in `client.ts`:
 
 ```ts
constructor(apiKey: string, blog: Blog, options: Partial<ClientOptions> = {}) {
  const {baseUrl = "https://rest.akismet.com"} = options;
  const url = baseUrl instanceof URL ? baseUrl.href : baseUrl;
  
  this.apiKey = apiKey;
  this.baseUrl = new URL(url.endsWith("/") ? url : `${url}/`);
  this.blog = blog;
  this.isTest = options.isTest ?? false;
  this.userAgent = options.userAgent ?? `${navigator.userAgent} | Akismet/${Client.#version}`;
}
```

We use `navigator` if we don't explicitly specify the optional `userAgent` but its `undefined` in server context / component.

Workaround is to explicitly mention the `userAgent`:
```ts
const akismetClient = new Client("MY_KEY", blog, {
  userAgent: 'Node.js | Akismet/1.0'
});
```

I suggest by default if navigator is not defined we fallback to using `Node.js` as the user agent. Seems intuitive enough.

Let me know if you agree and I can put up a PR, or you can yourself if you prefer.

Thanks!

**Comment by cedx** [comment]
Hi! Thanks for reporting this error. It happens because I tend to be ahead of the Node.js release schedule 😄 

`navigator.userAgent` was added to Node.js v21.1.0.
https://nodejs.org/api/globals.html#navigatoruseragent

That's why my package uses `>=22` as version constraint.
https://github.com/cedx/akismet.js/blob/main/package.json#L25

I assume you are using an LTS version (i.e. v20).

That said, it's not a problem to correct this.
I'll let you know as soon as I've published a fix (probably by the end of the weekend).
